src/main.cpp

```diff
@@ -264,4 +273,86 @@ void runSteeringDisplay() {
 	while (1) {
 		display.run();
 	}
+}
+
+void buttonRead(int buttonPress){
+	int ledPos;
+	for(int i = 0; i <8; i++){
+		shiftClk = 1;
+		wait_us(20);
+		shiftClk = 0;
+		shiftLatch = 1;
+
+		if(i == 3 && buttonPress == 1){
+			ledPos = 1;
+			ledOn(ledPos);
+		}
+
+		else if(i == 4 && buttonPress == 1){
+			ledPos = 2;
+			ledOn(ledPos);
+		}
+
+		else if(i == 5 && buttonPress == 1){
+			ledPos = 3;
+			ledOn(ledPos);
+		}
+
+		else if(i == 6 && buttonPress == 0){
+			ledPos = 4;
+			ledOn(ledPos);
+		}
+
+		else if(i == 6 && buttonPress == 1){
+			ledPos = 5;
+			ledOn(ledPos);
+		}
+
+		else if(i == 7 && buttonPress == 1){
+			ledPos = 6;
+			ledOn(ledPos);
+		}
+
+		wait_us(20);
+		shiftLatch = 0;
+	}
+}
+
+void ledOn(int ledPos){
+	for(int i = 0; i < 8; i++){
+		shiftClk = 1;
+		wait_us(20);
+		shiftClk = 0;
+		shiftLatch = 1;
+		wait_us(20);
+		shiftLatch = 0;
+
+		if(i == ledPos) ledOut = 1;
+	}
+}
+
+void buttonTest(int buttonPress){
```

Comment:
What's this function for? It's not called anywhere in your code

Reply:
This was just for testing, I was not seeing any response when most of the buttons were pressed except the wipers button (on pressing and releasing the wipers button, each LED turns on one by one, almost like it is traversing the 595 shift register for each output).